otelx/tracing.go

```diff
@@ -191,9 +192,13 @@ func newOTLPHTTPExporter(tc Config) (sdktrace.SpanExporter, error) {
 		otlptracehttp.WithTimeout(tc.OTLP.Timeout),
 	}
 
-	if _, err := url.Parse(tc.OTLP.Endpoint); err == nil && tc.OTLP.Endpoint != "" {
-		// NOTE: if a valid URL is not passed, it will fallback to the env vars defined by the otel SDK
-		opts = append(opts, otlptracehttp.WithEndpointURL(tc.OTLP.Endpoint))
+	// If an endpoint is provided and it is valid, use the endpoint.
+	// If the endpoint is invalid return an error.
+	// If no endpoint is provided, fallback to otel sdk env vars.
+	if endpoint, err := buildOTLPEndpoint(tc.OTLP); endpoint != "" {
```

Comment:
We are only checking if the endpoint is set in the if statement here. Which means if `buildOTLPEndpoint` returns an error when the endpoint is set that error will be ignored. Because for else won't apply since endpoint was set. I think instead of `else if` below we can just put the `if err != nil` here on line 199.

Reply:
Something like this, right
```
 	if endpoint, err := buildOTLPEndpoint(tc.OTLP); endpoint != "" {
 	 	if err != nil {
 	 		return nil, err
 	 	}
 		opts = append(opts, otlptracegrpc.WithEndpointURL(endpoint))
 	}
```

Reply:
Or ignore me entirely because endpoint here is what is returned not what is in the config struct. 🤦🏼‍♀️